pipeline/pipeline/backend.py

```diff
@@ -208,54 +195,28 @@ def copy_output(r):
                 resources['requests']['memory'] = task._memory
 
             j = batch.create_job(image=task._image if task._image else default_image,
-                                 command=['/bin/bash', '-c', defs + cmd],
+                                 command=['/bin/bash', '-c', cmd],
                                  parent_ids=parent_ids,
                                  attributes=attributes,
-                                 volumes=volumes,
-                                 resources=resources)
+                                 resources=resources,
+                                 input_files=copy_task_inputs,
+                                 output_files=copy_task_outputs)
             n_jobs_submitted += 1
 
             task_to_job_mapping[task] = j
             job_id_to_command[j.id] = defs + cmd
             if verbose:
                 print(f"Submitted Job {j.id} with command: {defs + cmd}")
 
-        def write_pipeline_outputs(r, dest):
-            if isinstance(r, InputResourceFile):
```

Comment:
I thought I had this separate for a reason from the copying above. I need to think further if this is kosher.

Reply:
I changed the semantics slightly. Now, [I eagerly copy the results out](https://github.com/hail-is/hail/pull/5605/files/35cf6f149ade56d790deb0ebf65d516110b42747#diff-2e8d9bac670a9022b6a3bedd33315b56R163) when a job is finished. You waited for the entire pipeline to finish before copying all pipeline outputs out.